### ISSUE TITLE
Redesign

### ISSUE THREAD

**Issue by mvl22** [cyclestreets/cyclescape#923]
This will be the master issue for the redesign work.

**Comment by mvl22** [comment]
Wireframes implementing the proposed changes are now at:
https://blog.cyclescape.org/images/ProposedCyclescapeChangesWireframes.pdf

**Comment by nikolai-b** [comment]
Looks great!  Some of these really make the pages clearer.  I've a few questions after first reading and some might be overly concerned about implementation but I'll put them here anyway:

Currently users can sign-up to be notified about issues based on their location.  Does this exist in the redesign?  Is this based the group's location which might have lots of issues? I have a suggestion below (*).

With 1c) I don't like the "My" and "All", if selecting "All" and then selecting new filters would that override the "My" filters?

I think it is clearer to: 
- store the filters in the browser so they are kept after reload
- offer the user to save the combination of filters giving a name so they can create their own
- where we offer to save the location we could also offer to notify them about new issues that match the current search (I'm thinking like eBay's saved searches).  That could replace the current user location notification and it includes tags (*)
- have a simple "Clear all filters" button (which would be the same as "All")
- (less sure about this) add the filters to the URL params and read from these params to create them (allows the URL to be shared and the back button in the browser to work)

With 2a) would we limit the map to only show issues in that paginated group of issues (and not all "cycle parking" issues)?  

With 2c the maps would get very busy, could we limit them to only the n most recently discussed issues?  In camcycle the [planning](https://camcycle.cyclescape.org/tags/planning) map would have over 500 issues.
Also with 2c what is the order of the tags?  I would have started with the most issues and gone to the least but would you want the order to be the minimum time of the most recently discussed issue grouped by tag?

6b) so marginal interest and above user priorities would change to a star?

What would happen to started issues, would we add that to the search 1c or a separate page?

**Comment by mvl22** [comment]
> Currently users can sign-up to be notified about issues based on their location. Does this exist in the redesign?

Yes. It should be little different to what we have now. The filters stuff is the same as setting areas currently, just that it becomes an optional step, which if not done defaults to the group's area.

(Added to it is filtering by tags as well, but that could be added later. There is an existing issue on that somewhere, as it was always intended to be in the original 2011 design but never got done.)

> With 1c) I don't like the "My" and "All", if selecting "All" and then selecting new filters would that override the "My" filters?

The intention is that the My/All tabs are just a quick way for people to view stuff outside the user's defined areas. All is exactly what we currently have as the discussions tab. We're merely moving the My Cyclescape and Discussions tabs into one UI. All doesn't set any database values and doesn't affect the filters - it's just a temporary 'quick look' button essentially.

> would we limit the map to only show issues in that paginated group of issues (and not all "cycle parking" issues)?

The tag page (i.e. the new 'Virtual issue' page) would be paginated if required. It would limit to the current group within the subdomain context, and on www would show all (subject to privacy).

> With 2c the maps would get very busy, could we limit them to only the n most recently discussed issues? In camcycle the planning map would have over 500 issues.

Good point. I think this is something for the designer to deal with. I could imagine it might have a filter control to show most recent, for instance.

> Also with 2c what is the order of the tags?

Not sure. Maybe most popular. Maybe most recently-updated i.e. similar to the current issues page.

> 6b) so marginal interest and above user priorities would change to a star?

Basically I'm thinking we replace the current 1-5 priorities model with star. But we might not even need to bother with this at all. The star would just show as highlighted somehow in the listing.

> What would happen to started issues, would we add that to the search 1c or a separate page?

Not quite sure what you mean.

**Comment by mvl22** [comment]
I think the database structure changes are merely:

- Removal of issues. Discussion threads stand alone, but have the same structure (tags, etc.) as before, but with the addition of geometry. We would need to copy the geometry in the issue into the discussion at the point of this migration.

- Geography of a user becomes optional, with the default being the group's area. It is already but it's never been clear to me what effect that has on My Cyclescape.

- Add to the current area filtering system the filtering by tag as well. This is an OR, i.e. My Cyclescape is not just areas but also matches any tags. This was always part of the original 2011 spec, but we never got round to it. We could get on with this now.

- Change 1-5 priority to flag yes/no. We could get on with this now.

The new screens I think are pretty much all view changes. I don't think there's much actual new logic needed. For instance, the virtual issue from tags screen, we already have as the tag page:
https://camcycle.cyclescape.org/tags/cycleparking

Data fixes manually:

- Tags would require some manual data fixing - we want to try to make things more human-readable, i.e. "Cycle parking" rather than "cycleparking", so that the tags page has a sensible title at the top of the page. We could add a compiled map of each issue now.

**Comment by georgio8** [comment]
I am preparing some detailed comments on the Wireframe pages, but this post is to discuss my concern about the wider impact of completely eliminating Issues in favour of tagged threads:

Up to now all the Cyclescape Issues have been public. This constitutes a significant body of Open Data and I took advantage of that to generate my [Issue Map](https://issuemap.cyclescape.org/) providing a geographical overview of the issues with extensive and simple-to-configure filtering.

I do want to maintain the Issue Map: - LCC activists and staff make considerable use of it, but my concern applies not only for the Issue Map but for _anything equivalent to it_ that provides an open view of current and past issues.
The Issue Map isn't just a London-specific tool. It is configurable by URL parameters to provide UI's that serve a range of different campaigning purposes.
To maintain that capability the following data needs to be available for each discussion thread:
-  A title
- A description
- An owner/originator
- Some relevant and important tags, including a specified set that causes inclusion in any partticular instance of the map.
- A deadline
- A relevant URL
- A location and a shape (multiple shapes?) (Exception: chat threads?)
- Public/private status
- An image
I can't at the moment see how that data will be systematically collected and made available via the API as it is at present.

**Comment by mvl22** [comment]
Issues would effectively become replaced by virtual issues - the tags. If the threads are public and have a tag, then you’d get exactly the same issue map as currently. The geometry of each discussion thread for a tag becomes the geometry for that overall tag.

The design decision of issues being public has always been somewhat problems in one respect - it makes it impossible for, for instance, a group to start a private discussion on a matter which should not yet be public, e.g. pre-discussions with a developer. The workaround has been for a committee thread to be connected to some random other issue, but that is very unsatisfactory. The new proposal of virtual issues being created from the combination of public threads seems more satisfactory from a privacy perspective.

I agree it’s important to main this from an open data perspective. Indeed, StreetFocus relies on it. This could be problematic in Cambridge, where most threads are private, but I think the way round this is for the group to be able to create a token which essentially authorises the specific purpose of making thread geometries public.

**Comment by nikolai-b** [comment]
So on 1c if we were on the All tag there would be no option to set filters?  If there was an option then would it just override the filters in "my"?
If you are on the "My" section and you change a filter than your notification setting would automatically change? If they don't automatically change how would I go back to what the notification (or database) setting was?
My suggestion was to make an active step of "Save search as" or "Notify me of new issues that match this search" and then we could have many user locations (the model is 1-1 but could easily be 1 user - many user locations) where some could be set to have email notifications and some not.  Then we could have "Group" and "All" as tabs that cannot be edited, a "My" tab that is migrated from the current "My Cyclescape" and other tabs of saved searches.
